packages/salesforcedx-vscode-core/package.json

```diff
@@ -307,7 +307,7 @@
         },
         {
           "command": "sfdx.lightning.rename",
-          "when": "sfdx:project_opened && resource =~ /.*/(lwc|aura)/[^/]+(/[^/]+\\.(html|css|js|xml|svg|cmp|app|design|auradoc))?$/"
+          "when": "sfdx:project_opened && resource =~ /.*/(lwc|aura)/.*(/[^/]+\\.(html|css|js|xml|svg|cmp|app|design|auradoc))?$/"
```

Comment:
Basically, show the command to the user on anything inside the /lwc/ and /aura/ directories that have these file extensions.  This includes the component subdirectory named `__tests__` and files inside there.

If the user comes up with their own folder structure, this function should still work to resolve the parent component directory.